pyproject.toml

```diff
@@ -1,4 +1,126 @@
-[tool.black]
+[build-system]
+requires = ["setuptools>=61.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "yasa"
+description = "YASA: Analysis of polysomnography recordings."
+readme = "README.rst"
+license = {text = "BSD (3-clause)"}
+authors = [
+    {name = "Raphael Vallat", email = "raphaelvallat9@gmail.com"},
+    {name = "Remington Mallett", email = "mallett.remy@gmail.com"},
+]
+maintainers = [
+    {name = "Raphael Vallat", email = "raphaelvallat9@gmail.com"},
+    {name = "Remington Mallett", email = "mallett.remy@gmail.com"},
+]
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "Operating System :: MacOS",
+    "Operating System :: POSIX",
+    "Operating System :: Unix",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+]
+dynamic = ["version"]
+requires-python = ">=3.8"
+dependencies = [
+    "numpy>=1.18.1",
+    "scipy",
+    "pandas",
+    "matplotlib",
+    "seaborn",
+    "mne>=1.3",
+    "numba>=0.57.1",
+    "antropy",
+    "scikit-learn",
+    "tensorpac>=0.6.5",
+    "pyriemann>=0.2.7",
+    "sleepecg>=0.5.0",
+    "lspopt",
+    "ipywidgets",
+    "joblib",
+    "lightgbm",
+]
+
+[project.optional-dependencies]
+test = [
+    "pytest>=6",
+    "pytest-cov",
+    # Ensure coverage is new enough for `source_pkgs`.
+    "coverage[toml]>=5.3",
+    "ruff"
+]
+docs = [
+    "sphinx>7.0.0",
+    "pydata_sphinx_theme",
+    "numpydoc",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "sphinx-notfound-page",
+]
+
+[project.urls]
+Homepage = "https://github.com/raphaelvallat/yasa/"
+Downloads = "https://github.com/raphaelvallat/yasa/"
+
+[tool.setuptools]
+py-modules = ["yasa"]
+include-package-data = true
+
+[tool.setuptools.package-data]
+yasa = [
+    "classifiers/*.joblib",
+]
+
+[tool.setuptools.packages.find]
+namespaces = false
+where = ["yasa"]
```

Comment:
100% modern packaging would include dropping into `src/yasa`.

Reply:
I know but I'd rather do that in a separate PR after we've merged https://github.com/raphaelvallat/yasa/pull/127

Reply:
Actually I went ahead and implemented this here as well. We'll rebase this one after #127 is merged